updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -83,44 +94,59 @@ def create_file_fetcher(directory: nil)
       # prefer credentials directly from the root of the file (will contain secrets) but if not specified, fall back to
       # the job's credentials-metadata that has no secrets
       credentials = job_definition.fetch("credentials", job_credentials_metadata)
+
+      # prefer credentials directly from the root of the file (will contain secrets) but if not specified, fall back to
+      # the job's credentials-metadata that has no secrets
+      # Convert experiments hash to have string keys and string values
+      experiments_options = job.experiments.transform_keys(&:to_s).transform_values(&:to_s)
+
       args = {
         source: job.source.clone.tap { |s| s.directory = directory_to_use },
         credentials: credentials,
-        options: job.experiments
+        options: experiments_options
       }
       args[:repo_contents_path] = Environment.repo_contents_path if job.clone? || already_cloned?
       Dependabot::FileFetchers.for_package_manager(job.package_manager).new(**args)
     end
 
     # The main file fetcher method that now calls the create_file_fetcher method
     # and ensures it uses the same repo_contents_path setting as others.
+    sig { returns(Dependabot::FileFetchers::Base) }
     def file_fetcher
-      @file_fetcher ||= create_file_fetcher
+      @file_fetcher ||= T.let(create_file_fetcher, T.nilable(Dependabot::FileFetchers::Base))
     end
 
     # This method is responsible for creating or retrieving a file fetcher
     # from a cache (@file_fetchers) for the given directory.
+    sig { params(directory: String).returns(Dependabot::FileFetchers::Base) }
     def file_fetcher_for_directory(directory)
+      @file_fetchers = T.let(@file_fetchers, T.nilable(T::Hash[String, Dependabot::FileFetchers::Base]))
       @file_fetchers ||= {}
       @file_fetchers[directory] ||= create_file_fetcher(directory: directory)
     end
 
     # rubocop:disable Metrics/PerceivedComplexity
-    def dependency_files_for_multi_directories
-      return @dependency_files_for_multi_directories if defined?(@dependency_files_for_multi_directories)
+    sig { returns(T.nilable(T::Array[Dependabot::DependencyFile])) }
+    def dependency_files_for_multi_directories # rubocop:disable Metrics/AbcSize,Metrics/CyclomaticComplexity
```

Comment:
We should try to avoid disabling cops. I know it's happened a lot in the past, but now it feels like the code is littered with disabled cops.